### PR TITLE
Added documentation for Otel metrics

### PR DESCRIPTION
**Pull request by JonasKunz**
## What does this PR do?
Adds the documentation for #2968 and #3014.

### CODE REVIEW

File: docs/api-opentelemetry.asciidoc [hunk 1]
```diff
@@ -7,7 +7,7 @@ endif::[]
 === OpenTelemetry bridge
 
 The Elastic APM OpenTelemetry bridge allows creating Elastic APM `Transactions` and `Spans`,
-using the OpenTelemetry API.
+using the OpenTelemetry API. OpenTelemetry metrics are collected and shipped to elastic as well.
```
Comment (SylvainJuge) [inline docs/api-opentelemetry.asciidoc:10]:
```suggestion
using the OpenTelemetry API. OpenTelemetry metrics are also collected.
```

`elastic` in itself is a bit vague destination, and we'd better avoid any confusion where the metrics are sent.

File: docs/api-opentelemetry.asciidoc [hunk 2]
```diff
@@ -136,17 +136,106 @@ try (Scope scope = custom.makeCurrent()) {
...
+import io.opentelemetry.api.metrics.Meter;
+
+Meter myMeter = GlobalOpenTelemetry.getMeter("my_meter");
+LongCounter counter = meter.counterBuilder("my_counter").build();
+counter.add(42);
+----
+
+We don't require you to setup an OpenTelemetry `MeterProvider` within `GlobalOpenTelemetry` yourself.
+The Elastic APM Java Agent will detect if no `MeterProvider` was configured and will provide its own automatically in this case.
+If you provide your own `MeterProvider` (see <<otel-metrics-sdk>>), it will not be overridden by the agent.
```
Comment (SylvainJuge) [inline docs/api-opentelemetry.asciidoc:172]:
```suggestion
If you provide your own `MeterProvider` (see <<otel-metrics-sdk>>), the agent will use the provided instance.
```

File: docs/api-opentelemetry.asciidoc [hunk 3]
```diff
@@ -136,17 +136,106 @@ try (Scope scope = custom.makeCurrent()) {
...
+
+[float]
+[[otel-metrics-sdk]]
+==== Using a customized MeterProvider
+
+In some cases using just the <<otel-metrics-api, OpenTelemetry API for metrics>> might not be flexible enough.
+Some example use cases are:
+
+ * You want to use OpenTelemetry Views (e.g. to customize histogram buckets on a per-metric basis)
+ * You want to export metrics to other tools in addition to Elastic APM (e.g. prometheus)
```
Comment (SylvainJuge) [inline docs/api-opentelemetry.asciidoc:182]:
```suggestion
 * Using OpenTelemetry Views (e.g. to customize histogram buckets on a per-metric basis)
 * Exporting metrics to other tools in addition to Elastic APM (e.g. prometheus)
```

File: docs/api-opentelemetry.asciidoc [hunk 4]
```diff
@@ -136,17 +136,106 @@ try (Scope scope = custom.makeCurrent()) {
...
+
+For these use cases you can just setup you OpenTelemetry SDK `MeterProvider`.
+The Elastic APM Agent will take care of installing an additional `MetricExporter` via instrumentation,
+which will ship the metric data to Elastic APM.
+This requires you to use OpenTelemetry version `1.16.0` or newer.
```
Comment (SylvainJuge) [inline docs/api-opentelemetry.asciidoc:187]:
```suggestion
This requires using OpenTelemetry version `1.16.0` or newer.
```
[question] : is the version constraint from our agent instrumentation or something that is not supported by older OTel versions ? If it is the latter, then I don't think it's worth mentioning it here (as it is already implicitly covered in the Otel docs).

File: docs/api-opentelemetry.asciidoc [hunk 5]
```diff
@@ -136,17 +136,106 @@ try (Scope scope = custom.makeCurrent()) {
...
 [[otel-caveats]]
 ==== Caveats
 Not all features of the OpenTelemetry API are supported.
 
-[float]
-[[otel-metrics]]
-===== Metrics
-This bridge only supports the tracing API.
-The Metrics API is currently not supported.
-
```
Comment (SylvainJuge) [inline docs/api-opentelemetry.asciidoc:149]: :heart_eyes:

**Review by SylvainJuge — APPROVED** [review]
LGTM, I mostly have a few suggestions for wording.

There is also a lot of "you", so I would suggest simplifying a bit to avoid duplication, also that's probably bothering me only because doing the same in French would imply that we try to be friends with the end-user, so I'm definitely biased.

File: docs/api-opentelemetry.asciidoc (hunk 4 above)
Comment (JonasKunz) [reply docs/api-opentelemetry.asciidoc:187]: Otel 1.16.0 added the capability for exporters to define the default aggregations to use (and temporality if I remember correctly). These features are used by our Otel Metric Exporter. For this reason the instrumentation ignores otel-SDKs which are older than 1.16.
For the API-based usage we support all opentelemetry versions (metrics were added in 1.10 )

**Comment by JonasKunz** [comment]
Great suggestions, thanks!

**Pull request merged**